src/main/java/org/candlepin/subscriptions/tally/TallySnapshotController.java

```diff
@@ -90,12 +96,21 @@ public TallySnapshotController(
 
   @Timed("rhsm-subscriptions.snapshots.single")
   public void produceSnapshotsForAccount(String account) {
+    // NOTE: This lookup should happen when the process starts and should be passed
+    //       along. This should be removed once the task processor is updated to use
+    //       the org_id.
+    Optional<AccountConfig> accountLookup = accountRepo.findById(account);
+    if (accountLookup.isEmpty()) {
+      throw new IllegalArgumentException(
+          String.format("Invalid account number: %s. Has the account been opted in?", account));
+    }
+
```

Comment:
Just a suggestion, but I find this syntax a little easier conceptually and it avoids the `get` call later on.
```suggestion
    AccountConfig accountLookup = accountRepo.findById(account).orElseThrow(
        () -> new IllegalStateException(
        String.format("Invalid account number: %s. Has the account been opted in?", account))
    );
```

Reply:
DONE